dlk/python/dlk/templates/manual/consts/input.tpl.cpp

```diff
@@ -28,19 +28,11 @@ const TensorView<{{ node.dtype.cpptype() }}, MemoryLayout::Atom>
 
 {% if node.transposed_data %}
 
-#if defined(RUN_ON_FPGA)
 static Base<{{ node.dtype.cpptype() }}>::type {{ node.name }}_raw[] = {
   {% for d in node.transposed_data -%}
   {{- d -}},
   {%- endfor %}
 };
-#else
-static Base<{{ node.dtype.cpptype() }}>::type {{ node.name }}_raw[] = {
-  {% for d in node.data.flatten() -%}
-  {{- d -}},
-  {%- endfor %}
-};
-#endif
```

Comment:
FPGA uses a layout of kernel data which is different from CPU version...
Do you really want to remove this one? :thinking:

Reply:
This PR code convert kernel layout in runtime from layout for FPGA to layout for CPU.

Reply:
Could not say that converting layout of statically generated data sounds like a good idea to me... :thinking:

Reply:
Kernel layout for kn2row is not same as kernel layout of tiling/im2col, if we want to generate data statically, python code should know which method is used...

Reply:
Hmm...
Doesn't seem particularly challenging to do it in python, but as workaround sounds good for me...

On the other hand, not sure do we really need `im2col`, btw. So far we did not use quantized convolutions which kn2row (read fpga) don't work for, and if we use them performance will be not very good. Thus maybe it is a good time to stop pretending to be generic. :thinking:

Reply:
OK. I'll remove `im2col` and `kn2row` and generate kernel data for CPU statically.

Reply:
It could take several days to implement tiling for x86(done in another PR).

Reply:
Could we use `./dlk/python/dlk/templates/src/func/impl/generic/quantized_conv2d_kn2row.cpp` for x86 for now? :thinking:

Reply:
It's still good (more layout convert in python needed).